internal/infrastructure/kubernetes/proxy_deployment.go

```diff
@@ -0,0 +1,297 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package kubernetes
+
+import (
+	"context"
+	// Register embed
+	_ "embed"
+	"fmt"
+	"strings"
+	"text/template"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/pointer"
+
+	egcfgv1a1 "github.com/envoyproxy/gateway/api/config/v1alpha1"
+	"github.com/envoyproxy/gateway/internal/envoygateway/config"
+	"github.com/envoyproxy/gateway/internal/gatewayapi"
+	"github.com/envoyproxy/gateway/internal/ir"
+	"github.com/envoyproxy/gateway/internal/provider/utils"
+	xdsrunner "github.com/envoyproxy/gateway/internal/xds/server/runner"
+)
+
+const (
+	// envoyContainerName is the name of the Envoy container.
+	envoyContainerName = "envoy"
+	// envoyNsEnvVar is the name of the Envoy Gateway namespace environment variable.
+	envoyNsEnvVar = "ENVOY_GATEWAY_NAMESPACE"
+	// envoyPodEnvVar is the name of the Envoy pod name environment variable.
+	envoyPodEnvVar = "ENVOY_POD_NAME"
+	// envoyCfgFileName is the name of the Envoy configuration file.
+	envoyCfgFileName = "bootstrap.yaml"
+	// envoyHTTPPort is the container port number of Envoy's HTTP endpoint.
```

Comment:
@danehans is `envoyHTTPPort` and `envoyHTTPSPort` required ? doesnt the gateway API layer decide the infra ports ?

Reply:
thanks for analyzing, will raise a PR to clean this up